Several quarkus integration tests fail to compile to native with latest GraalVM master

### Describe the bug

`oidc-client` is one, `main` integration-test is another. Fails the same way with JDK 17-based build as well as JDK 20-base build.

It looks like `ResourceCleaner.CLEANER` from reasteasy becomes newly? reachable? `native-image` generation fails with:
```
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building native image source jar: /home/runner/work/mandrel/mandrel/quarkus/integration-tests/oidc-client/target/quarkus-integration-test-oidc-client-999-SNAPSHOT-native-image-source-jar/quarkus-integration-test-oidc-client-999-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Building native image from /home/runner/work/mandrel/mandrel/quarkus/integration-tests/oidc-client/target/quarkus-integration-test-oidc-client-999-SNAPSHOT-native-image-source-jar/quarkus-integration-test-oidc-client-999-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on native-image 23.0.0-devc2e5c2c9 Mandrel Distribution (Java Version 17.0.7-beta+3-202302240048)
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] /home/runner/work/mandrel/mandrel/openjdk/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dlogging.initial-configurator.min-level=400 -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.noUnsafe=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=3 -J-Duser.language=en -J-Duser.country=US -J-Dfile.encoding=UTF-8 --features=io.quarkus.runner.Feature,io.quarkus.runtime.graal.DisableLoggingFeature -J--add-exports=java.security.jgss/sun.security.krb5=ALL-UNNAMED -J--add-opens=java.base/java.text=ALL-UNNAMED -J--add-opens=java.base/java.io=ALL-UNNAMED -J--add-opens=java.base/java.lang.invoke=ALL-UNNAMED -J--add-opens=java.base/java.util=ALL-UNNAMED -H:BuildOutputJSONFile=quarkus-integration-test-oidc-client-999-SNAPSHOT-runner-build-output-stats.json -H:+AllowFoldMethods -J-Djava.awt.headless=true --no-fallback --link-at-build-time -H:+ReportExceptionStackTraces -J-Xmx5g -H:-AddAllCharsets --enable-url-protocols=http,https -H:NativeLinkerOption=-no-pie -H:-UseServiceLoaderFeature -H:+StackTrace -J--add-exports=org.graalvm.sdk/org.graalvm.nativeimage.impl=ALL-UNNAMED --exclude-config io\.netty\.netty-codec /META-INF/native-image/io\.netty/netty-codec/generated/handlers/reflect-config\.json --exclude-config io\.netty\.netty-handler /META-INF/native-image/io\.netty/netty-handler/generated/handlers/reflect-config\.json quarkus-integration-test-oidc-client-999-SNAPSHOT-runner -jar quarkus-integration-test-oidc-client-999-SNAPSHOT-runner.jar
========================================================================================================================
GraalVM Native Image: Generating 'quarkus-integration-test-oidc-client-999-SNAPSHOT-runner' (executable)...
========================================================================================================================
Warning: Could not register io.quarkus.oidc.client.filter.runtime.AbstractOidcClientRequestFilter_nondefaultclient for serialization as it does not implement Serializable.
[1/8] Initializing...                                                                                   (14.3s @ 0.22GB)
 Version info: 'GraalVM 23.0.0-devc2e5c2c9 Java 17.0.7-beta+3-202302240048 Mandrel Distribution'
 Java version info: '17.0.7-beta+3-202302240048'
 C compiler: gcc (linux, x86_64, 11.3.0)
 Garbage collector: Serial GC (max heap size: unlimited)
 3 user-specific feature(s)
 - io.quarkus.runner.Feature: Auto-generated class by Quarkus from the existing extensions
 - io.quarkus.runtime.graal.DisableLoggingFeature: Disables INFO logging during the analysis phase
 - org.eclipse.angus.activation.nativeimage.AngusActivationFeature
[2/8] Performing analysis...  [******]                                                                 (121.6s @ 2.32GB)
  14,722 (89.06%) of 16,530 types reachable
  23,726 (63.19%) of 37,545 fields reachable
  76,107 (60.10%) of 126,631 methods reachable
   4,574 types,   176 fields, and 2,938 methods registered for reflection
      63 types,    68 fields, and    55 methods registered for JNI access
       4 native libraries: dl, pthread, rt, z

Error: Detected a java.lang.ref.Cleaner object in the image heap which uses a daemon thread that invokes cleaning actions, but threads running in the image generator are no longer running at image runtime.  To see how this object got instantiated use --trace-object-instantiation=java.lang.ref.Cleaner. The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Detailed message:
Trace: Object was reached by
  trying to constant fold static field org.jboss.resteasy.spi.ResourceCleaner.CLEANER
    at org.jboss.resteasy.spi.ResourceCleaner.register(ResourceCleaner.java:48)
  parsing method org.jboss.resteasy.spi.ResourceCleaner.register(ResourceCleaner.java:48) reachable via the parsing context
    at static root method.(Unknown Source)


------------------------------------------------------------------------------------------------------------------------
com.oracle.svm.core.util.UserError$UserException: Detected a java.lang.ref.Cleaner object in the image heap which uses a daemon thread that invokes cleaning actions, but threads running in the image generator are no longer running at image runtime.  To see how this object got instantiated use --trace-object-instantiation=java.lang.ref.Cleaner. The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Detailed message:
Trace: Object was reached by
  trying to constant fold static field org.jboss.resteasy.spi.ResourceCleaner.CLEANER
    at org.jboss.resteasy.spi.ResourceCleaner.register(ResourceCleaner.java:48)
  parsing method org.jboss.resteasy.spi.ResourceCleaner.register(ResourceCleaner.java:48) reachable via the parsing context
    at static root method.(Unknown Source)


	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:85)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:248)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:821)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:605)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:562)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:408)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:616)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:134)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:94)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Detected a java.lang.ref.Cleaner object in the image heap which uses a daemon thread that invokes cleaning actions, but threads running in the image generator are no longer running at image runtime.  To see how this object got instantiated use --trace-object-instantiation=java.lang.ref.Cleaner. The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Detailed message:
Trace: Object was reached by
  trying to constant fold static field org.jboss.resteasy.spi.ResourceCleaner.CLEANER
    at org.jboss.resteasy.spi.ResourceCleaner.register(ResourceCleaner.java:48)
  parsing method org.jboss.resteasy.spi.ResourceCleaner.register(ResourceCleaner.java:48) reachable via the parsing context
    at static root method.(Unknown Source)


	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:126)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:816)
	... 6 more
                        7.2s (5.2% of total time) in 27 GCs | Peak RSS: 3.12GB | CPU load: 1.94
------------------------------------------------------------------------------------------------------------------------
```

### Expected behavior

Native image generation passes.

### Actual behavior

Fails as above.

### How to Reproduce?

Run integration-tests `main` or `oidc-client` with a GraalVM/Mandrel build from latest master. E.g. from here:
https://github.com/graalvm/mandrel/suites/11215681139/artifacts/573900606

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17.0.7+1

### GraalVM version (if different from Java)

lastest master

### Quarkus version or git rev

main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

See also:
https://github.com/graalvm/mandrel/actions/runs/4278105893/jobs/7447740912#step:11:324

--- a/extensions/resteasy-classic/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java
+++ b/extensions/resteasy-classic/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java
@@ -56,6 +56,7 @@ import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
 import io.quarkus.deployment.builditem.StaticInitConfigBuilderBuildItem;
 import io.quarkus.deployment.builditem.SystemPropertyBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
 import io.quarkus.deployment.util.ServiceUtil;
 import io.quarkus.resteasy.common.runtime.ResteasyInjectorFactoryRecorder;
@@ -341,6 +342,16 @@ public class ResteasyCommonProcessor {
     void registerNativeImageResources(BuildProducer<ServiceProviderBuildItem> serviceProvider) {
         serviceProvider.produce(ServiceProviderBuildItem
                 .allProvidersFromClassPath(org.jboss.resteasy.spi.config.ConfigurationFactory.class.getName()));
+    }
+
+    /**
+     * ResourceCleaner contains java.lang.ref.Cleaner references which need to get
+     * runtime initialized.
+     */
+    @BuildStep
+    public RuntimeInitializedClassBuildItem runtimeInitResourceCleaner() {
+        return new RuntimeInitializedClassBuildItem(
+                "org.jboss.resteasy.spi.ResourceCleaner");
     }
 
     private void registerJsonContextResolver(

